File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_csharp/src_wrapper/GnDspFeature.cs

namespace GracenoteSDK {

/**************************************************************************
** GnDspFeature
*/
public class GnDspFeature : global::System.IDisposable {
  private global::System.Runtime.InteropServices.HandleRef swigCPtr;
  protected bool swigCMemOwn;

  internal GnDspFeature(global::System.IntPtr cPtr, bool cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = new global::System.Runtime.InteropServices.HandleRef(this, cPtr);
  }

  internal static global::System.Runtime.InteropServices.HandleRef getCPtr(GnDspFeature obj) {
    return (obj == null) ? new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero) : obj.swigCPtr;
  }

  ~GnDspFeature() {
    Dispose();
  }

  public virtual void Dispose() {
    lock(this) {
      if (swigCPtr.Handle != global::System.IntPtr.Zero) {
        if (swigCMemOwn) {
          swigCMemOwn = false;
          gnsdk_csharp_marshalPINVOKE.delete_GnDspFeature(swigCPtr);
        }
        swigCPtr = new global::System.Runtime.InteropServices.HandleRef(null, global::System.IntPtr.Zero);
      }
      global::System.GC.SuppressFinalize(this);
    }
  }

  public string FeatureData() {
    string ret = gnsdk_csharp_marshalPINVOKE.GnDspFeature_FeatureData(swigCPtr);
    return ret;
  }

  public GnDspFeatureQuality FeatureQuality() {
    GnDspFeatureQuality ret = (GnDspFeatureQuality)gnsdk_csharp_marshalPINVOKE.GnDspFeature_FeatureQuality(swigCPtr);
    return ret;
  }

}

}

File: gnsdk-3.08.6.5437o-20170914/wrappers/gnsdk_java/src_wrapper/j2se/com/gracenote/gnsdk/GnDspFeature.java

package com.gracenote.gnsdk;

/************************************************************************** 
** {@link GnDspFeature} 
*/ 
 
public class GnDspFeature {
  private transient long swigCPtr;
  protected transient boolean swigCMemOwn;

  protected GnDspFeature(long cPtr, boolean cMemoryOwn) {
    swigCMemOwn = cMemoryOwn;
    swigCPtr = cPtr;
  }

  protected static long getCPtr(GnDspFeature obj) {
    return (obj == null) ? 0 : obj.swigCPtr;
  }

  protected void finalize() {
    delete();
  }

  public synchronized void delete() {
    if (swigCPtr != 0) {
      if (swigCMemOwn) {
        swigCMemOwn = false;
        gnsdk_javaJNI.delete_GnDspFeature(swigCPtr);
      }
      swigCPtr = 0;
    }
  }

  public String featureData() {
    return gnsdk_javaJNI.GnDspFeature_featureData(swigCPtr, this);
  }

  public GnDspFeatureQuality featureQuality() {
    return GnDspFeatureQuality.swigToEnum(gnsdk_javaJNI.GnDspFeature_featureQuality(swigCPtr, this));
  }

}
